var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -31,6 +34,12 @@ class Esmf(MakefilePackage):
     variant("netcdf", default=True, description="Build with NetCDF support")
     variant("pnetcdf", default=True, description="Build with pNetCDF support")
     variant("xerces", default=True, description="Build with Xerces support")
+    variant(
+        "external-pio",
```

Comment:
"External" has another meaning in Spack (something installed on the system, outside of Spack's control). Can you change the name to:
```suggestion
        "pio",
```
?

Reply:
I was following the pattern already found in the package, there is an "external-lapack" entry at line 33.  I'm not sure I follow the suggestion you are making - can you elaborate?   @climbfuji can you weigh in on the use of "external-" in the esmf package file?

Reply:
> can you elaborate?

If Spack doesn't manage the entire stack it cannot guarantee there won't be conflicts in the DAG, because the software has a dependency that is not modeled. Whenever possible we advise packagers to rely on another Spack package as a dependency, rather than use something vendored. Since this package has an option to choose between vendored and non-vendored for `parallelio`, it means that it can rely on Spack built dependencies. Therefore I think we should remove the option to build with the vendored one to avoid the problematic case.

Reply:
I agree but this is not my package, I am not in a position to make that decision.   @rsdunlapiv

Reply:
Also this option is only available in esmf > 8.3.0

Reply:
Yes, we should allow Spack to manage the ParallelIO dependency and the default for ESMF should be to have ParallelIO as a required dependency.  A variant should allow turning it off in which case ESMF_PIO should be OFF.  In Spack environments, I don't think we would want to support ESMF_PIO=internal given comments from @alalazo. 

We should consistently use "ParallelIO" (not PIO) if that is the name of the dependent Spack package and remove the "external-" prefix in the variant.

@theurich @uturuncoglu

Reply:
I have also made couple of changes in the ESMF package and it would be nice to add them to this PR. Here is the comparison,

https://github.com/NOAA-EMC/spack/compare/jcsda_emc_spack_stack...uturuncoglu:spack:hotfix/esmf

I just change the default for xerces variant (it is now False) and change the pnetcdf to use pnetcdf-config (this will allow to appear pnetcdf related definitions in the esmf.mk file).

Reply:
@uturuncoglu I have added your changes to the PR.

Reply:
@jedwards4b thanks. that is great! we still need to find a way for patches in case of using develop. I was thinking that running `make info` before applying patch could be the solution to get used version or using same way that is used in here to return `ESMF_VERSION_STRING_GIT ` `https://github.com/esmf-org/esmf/blob/15670f87e16db8f6159356828c77fa786e27752d/makefile#L77`. BTW, have you tried to build develop branch using spark? I tried yesterday and it was failing due to this issue.